snippets/deepClone.md

```diff
@@ -12,7 +12,7 @@ const deepClone = obj => {
   Object.keys(clone).forEach(
     key => (clone[key] = typeof obj[key] === 'object' ? deepClone(obj[key]) : obj[key])
   );
-  return Array.isArray(obj) ? (clone.length = obj.length) && Array.from(clone) : clone;
+  return Array.isArray(obj) ? Array.from({ length: obj.length }) : clone;
```

Comment:
deepClone({ a: [] }); // expect output { a: [] } not { a: 0 }

Reply:
```js
deepClone({ a: [1,2,3] }) // { a: []}
```

Reply:
I think there's no really elegant way to fix the edge case, so we probably need to check if it's an array and empty to return and empty array, otherwise return the clone. I am trying to solve this in my head, but I probably need to write the code to see if there's a short way to write this. Does `Array.from(otherArray)` work or `Object.assign([],otherArray)` work? Those are the only things that come to mind apart from the spread operator.